src/library/tLuaCOMTypeHandler.cpp

```diff
@@ -1359,11 +1360,19 @@ stkIndex tLuaCOMTypeHandler::get_from_array(lua_State* L,
 
   HRESULT hr = SafeArrayGetElement(safearray, indices, pv);
 
-  if(FAILED(hr))
+  if(FAILED(hr)) {
+    if (vt != VT_VARIANT) {
+      VariantClear(&varg);
+    }
     LUACOM_EXCEPTION(INTERNAL_ERROR);
+  }
 
   com2lua(L, varg);
 
+  if (vt != VT_VARIANT) {
```

Comment:
I have cocerns here - may be `VariantClear` shold be called without comdition.
@1linux can you easily test with safearray of variants?